module/data/item/spell.mjs

```diff
@@ -157,7 +157,9 @@ export default class SpellData extends ItemDataModel.mixin(
 
   /** @inheritdoc */
   get _typeAbilityMod() {
-    return this.parent?.actor?.system.attributes.spellcasting || "int";
+    return this.parent?.actor?.spellcastingClasses[this.sourceClass]?.spellcasting.ability
```

Comment:
`Actor5e#spellcastingClasses` should probably be cached in the same way `Actor5e#classes` is, since it's going to be called by every single spell Item during preparation, potentially more than once.